commands_add.go

```diff
@@ -46,10 +46,31 @@ func (a addCmd) ToAuditEntry() string {
 		<username>%s</username>
 		<funds>%.02f</funds>
 	</userCommand>`,
-		time.Now().UnixNano()/1e6, redisBaseKey, a.id, a.userID, a.amount.ToFloat(),
+		time.Now().UnixNano()/1e6, redisBaseKey, a.txID, a.userID, a.amount.ToFloat(),
 	)
 }
 
 func (a addCmd) Execute() {
-	consoleLog.Warning("Not implemented: ADD")
+	// Create an account if one does not exist
+	if !accountStore.HasAccount(a.userID) {
+		consoleLog.Noticef("Creating account for %s", a.userID)
+		if err := accountStore.CreateAccount(a.userID); err != nil {
+			consoleLog.Error(err.Error())
+			abortTx("Account creation failed")
+		}
+	}
+
+	// Add the amounts
+	consoleLog.Infof("Adding %s to %s", a.amount, a.userID)
```

Comment:
Good catch, the user is the key so this doesn't work.

Motivation for it is to provide a full trace for account state if (when) things start behaving strangely. Being able to invoke `-l DEBUG` has saved my butt a few times in the lab.